packages/react-core/src/demos/PasswordStrength.md

```diff
@@ -0,0 +1,170 @@
+---
+id: Password strength
+section: demos
+---
+
+import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
+
+## Demos
+
+### Basic validation
+
+Note, the validation and password strength rules are only examples, demonstrating the changes in the UI when certain conditions are met. We expect consumers will substitute their own, more robust, validation algorithm. In this demo the password strength is determined by how often validation rules are met. A good open-source password strength estimator, recommended by InfoSec, can be found here: https://github.com/dropbox/zxcvbn
+
+```js
+import React from 'react';
+import {
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  Popover,
+  HelperTextItem,
+  TextInput
+} from '@patternfly/react-core';
+import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
+
+class PasswordStrengthDemo extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: '',
+      ruleLength: 'indeterminate',
+      ruleContent: 'indeterminate',
+      ruleCharacters: 'indeterminate',
+      passStrength: { variant: 'error', icon: <ExclamationCircleIcon />, text: 'Weak' }
+    };
+
+    this.handlePasswordInput = password => {
+      this.setState({ password });
+      this.validate(password);
+    };
+
+    this.validate = password => {
+      if (password === '') {
+        this.setState({
+          ruleLength: 'indeterminate',
+          ruleContent: 'indeterminate',
+          ruleCharacters: 'indeterminate'
+        });
+        return;
+      }
+
+      if (password.length < 14) {
+        this.setState({ ruleLength: 'error' });
+      } else {
+        this.setState({ ruleLength: 'success' });
+      }
+
+      if (/redhat/gi.test(password) || /red/gi.test(password) || /hat/gi.test(password) || /rh/gi.test(password)) {
```

Comment:
@mcarrano fwiw, the validation here checks for "redhat", "red", "hat", and "rh". So the password validation fails with words that contain those strings, which seems confusing.

<img width="382" alt="Screen Shot 2021-07-29 at 4 03 25 PM" src="https://user-images.githubusercontent.com/35148959/127565531-c39418dc-dd2b-485f-a7c8-6c450afa380b.png">
<img width="406" alt="Screen Shot 2021-07-29 at 4 03 16 PM" src="https://user-images.githubusercontent.com/35148959/127565533-e186fd35-5162-4ab7-bfc1-f86ad6678c4d.png">
<img width="363" alt="Screen Shot 2021-07-29 at 4 05 23 PM" src="https://user-images.githubusercontent.com/35148959/127565682-1808cd3e-305c-4d5e-b61c-a9327dbb4d83.png">

Any problem with updating the helper text to say `Cannot contain the word "redhat"` instead of `Cannot contain any variation of the word "redhat"`?

Reply:
@mcarrano Updated the second validation rule to 'cannot contain the word redhat' instead of variations. Let me know if that's okay or if I should change it back.

Reply:
Yes, this is fine @kmcfaul . I just noticed a typo in your helper text string, however. It reads, "Cannot contain any the word "redhat" Please remove the work "any." Thanks.